### DISCUSSION TITLE
What is the clustering method in plotMarkerHeatmap and how are pseudo-bulk profiles generated?

### DISCUSSION THREAD

**Discussion by RegnerM2015** [GreenleafLab/ArchR#892] [Questions / Documentation]
I would like find out if hierarchical clustering is used in the generation of heatmaps for plotMarkerHeatmap. If so, what linkage method is used? 

Also, how are the cells within a given cell type aggregated to form one profile for each column in the heatmap? Is it a sum of counts or an average of counts?

**Comment by rcorces** [comment]
https://github.com/GreenleafLab/ArchR/blob/968e4421ce7187a8ac7ea1cf6077412126876d5f/R/ArchRHeatmap.R#L151

I would look up the documentation in ComplexHeatmap if you want to know more. ArchR uses Ward D2 hierarchical clustering as the default and we have not provided an easy way to change this but could implement it if there is a good rationale.

**Reply by RegnerM2015** [reply]
Sorry I missed that line! Thanks for the help!

Could you point me in the right direction of the code for how the cell cluster aggregation is done for making the columns in the heatmap?

**Reply by rcorces** [reply]
I believe it is a sum, computed after normalization based on `normBy`.

**Reply by RegnerM2015** [reply]
Awesome, thank you!
